core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -103,6 +104,81 @@ private[spark] class AppStatusListener(
     }
   }
 
+  // visible for tests
+  private[spark] def recoverLiveEntities(): Unit = {
+    if (!live) {
```

Comment:
This may need to cooperate with SPARK-28594's config.

Reply:
Hmm... I've found we don't recover AppStatusSource as it will not be passed when live == false. If possible I'd check whether KVStore is empty when live == true then, but it depends on the possibility. If not, we may have to live with.

Reply:
> If possible I'd put assertion whether KVStore is empty when live == true then, but it depends on the possibility. 

I think our current usage of a live(true) `AppStatusListener` guarantees the empty KVStore at the initialization step. So I don't understand well for `depends on the possibility` ? Do I miss something ?

And yet, it seems that we don't have `isEmpty` api for KVStore. Otherwise, I could put an assertion to reach a compromises between us.

Reply:
> I think our current usage of a live(true) AppStatusListener guarantees the empty KVStore at the initialization step. So I don't understand well for depends on the possibility ?

That's a "context" we might have a chance to break eventually and as a side-effect it will break here. I'm in favor of doing defensive programming: if there're preconditions it should be mentioned anywhere or asserted. But I agree we don't have isEmpty api for KVStore - let's leave it as it is.